src/librbd/ImageWatcher.cc

```diff
@@ -1516,6 +1516,15 @@ void ImageWatcher<I>::handle_rewatch_complete(int r) {
   CephContext *cct = m_image_ctx.cct;
   ldout(cct, 10) << this << " " << __func__ << ": r=" << r << dendl;
 
+  if (r < 0) {
+    lderr(cct) << this << " rewatch error: " << cpp_strerror(r) << dendl;
+    std::shared_lock owner_locker{m_image_ctx.owner_lock};
```

Comment:
I would combine the `owner_lock` blocks into one:
```
  {
    std::shared_lock owner_locker{m_image_ctx.owner_lock};
    if (r < 0) {
      lderr(cct) << this << " rewatch error: " << cpp_strerror(r) << dendl;
      if (m_image_ctx.exclusive_lock != nullptr) {
        // if stalled in request state machine -- abort
        m_image_ctx.exclusive_lock->handle_peer_notification(0);
      }
      return;
    }

    if (m_image_ctx.exclusive_lock != nullptr) {
      // update the lock cookie with the new watch handle
      m_image_ctx.exclusive_lock->reacquire_lock(nullptr);
    }
  }
```

Reply:
Good point! Will do in the next iteration.